lib/plotting_utils.py

```diff
@@ -0,0 +1,830 @@
+    if isinstance(cmap_case, str):
+        if (cmap_case not in plt.colormaps()) and (cmap_case not in ncl_defaults):
+            msg += f"\n\t\tInvalid cmap '{cmap_case}' for {plotty}, defaulting to {default_cmap}"
+            cmap_case = default_cmap
+    
+    adfobj.debug_log(msg)
+
+    return cmap_case
+
+
+# Conour Plot Prep Functions
+#----------------------------
+def resolve_hemi_level(adfobj, data, kwargs, polar_names):
+    """Resolve hemisphere and/or vertical level specific values from a dict."""
+    msg = f"{script_name}: resolve_hemi_level()"
+    hemi = kwargs.get("hemi")
+    lev = kwargs.get("lev")
+
+    if hemi and polar_names.get(hemi) in data:
+        hemi_data = data[polar_names[hemi]]
+        if isinstance(hemi_data, dict) and lev in hemi_data:
+            msg += f"\n\tPolar {hemi} with vertical level {lev}"
+            adfobj.debug_log(msg)
+            return hemi_data[lev]
+        msg += f"\n\tPolar {hemi} without vertical levels"
+        adfobj.debug_log(msg)
+        return hemi_data
+    elif lev and lev in data:
+        msg += f"\n\tVertical level {lev}"
+        adfobj.debug_log(msg)
+        return data[lev]
+
+    adfobj.debug_log(msg)
+    return None
+
+
+
+def resolve_levels(adfobj, plotty, plot_type_dict, kwargs, polar_names):
+        """Resolve contour levels based on user input and defaults."""
+        levels1 = None
+        msg = f"{script_name}: resolve_levels()"
+
+        # Map keys based on plot type
+        key_map = {
+            "diff": ("diff_contour_levels", "diff_contour_range", "diff_contour_linspace"),
+            "case": ("contour_levels", "contour_levels_range", "contour_levels_linspace"),
+        }
+        contour_levels, contour_range, contour_linspace = key_map.get(plotty, (None, None, None))
+
+        def process_entry(entry, kind, msg):
+            msg += f"\n\tprocess_entry()"
+            """Handle lists and dicts for levels/ranges/linspace."""
+            if isinstance(entry, list):
+                if len(entry) == 3:
+                    if kind == "range":
+                        msg += f"\n\tLevels specified for {plotty}: numpy.arange."
+                        #adfobj.debug_log(msg)
+                        return np.arange(*entry), msg
+                    elif kind == "linspace":
+                        msg += f"\n\tLevels specified for {plotty}: numpy.linspace."
+                        #adfobj.debug_log(msg)
+                        return np.linspace(*entry), msg
+                    else:
+                        msg += f"\n\tLevels specified for {plotty} as list of 3 values, please add more values."
+                        msg += " Will get contrours from data range."
+                        #adfobj.debug_log(msg)
+                        return None, msg
+                elif len(entry) < 3:
+                    msg += f"\n\tNot enough {kind} entries for {plotty} (<3) — ambiguous"
+                    #adfobj.debug_log(msg)
```

Comment:
Remove or un-comment the `debug_log` call here?  This same request holds for the other commented-out usages of `debug_log` below as well.